ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/Cleaner.java

```diff
@@ -334,7 +337,7 @@ private boolean removeFiles(String location, ValidWriteIdList writeIdList, Compa
     extraDebugInfo.setCharAt(extraDebugInfo.length() - 1, ']');
     LOG.info(idWatermark(ci) + " About to remove " + filesToDelete.size() +
          " obsolete directories from " + location + ". " + extraDebugInfo.toString());
-    if (filesToDelete.size() < 1) {
+    if (filesToDelete.size() < 1 && !isDynPartAbort(table, ci)) {
```

Comment:
Do we trust that the min write id is high enough that all aborted directories will be cleared?